microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py

```diff
@@ -148,13 +212,45 @@ def _send_subtest_msg(
         environment: Environment,
         test_name: str,
         test_status: TestStatus,
+        test_message: str = "",
     ) -> None:
         subtest_msg = create_test_result_message(
-            SubTestMessage,
-            test_id,
-            environment,
-            test_name,
-            test_status,
+            SubTestMessage, test_id, environment, test_name, test_status, test_message
         )
 
         notifier.notify(subtest_msg)
+
+    def _list_perf_metrics_tests(self, hypervisor: str = "kvm") -> List[str]:
+
+        tests_list = []
+        result = self.run(
+            f"tests --hypervisor {hypervisor} --metrics -- -- --list-tests",
+            timeout=self.TIME_OUT,
+            force_run=True,
+            cwd=self.repo_root,
+            shell=True,
+            expected_exit_code=0,
+        )
+
+        stdout = result.stdout
+
+        # Ex. String for below regex
+        # "boot_time_ms" (test_timeout=2s,test_iterations=10)
+        regex = '\\"(.*)\\" \\('
+
+        pattern = re.compile(regex)
+        tests_list = pattern.findall(stdout)
+
+        self._log.debug(f"Testcases found: {tests_list}")
+        return tests_list
+
+    def _process_perf_metric_test_result(self, output: str) -> str:
+        cnt = 0
+        for line in output.split("\n"):
+            if line.find("git_human_readable") >= 0:
+                cnt -= 1
+                break
+            cnt += 1
+
+        result = "\n".join([i.strip() for i in output.split("\n")[cnt:-5]])
```

Comment:
1. Give an example input above the parsing logic. It can help understand the logic and prevent break when it's refined.
2. Regex is easier to maintain and robust than using offset and split.

Reply:
Done